### PR TITLE
SUBMARINE-617. Update the apache/notebook:jupyter-notebook docker image

### PR DESCRIPTION
**Pull request by lowc1012**
### What is this PR for?
1. Update the apache/notebook:jupyter-notebook docker image
2. Preinstall notebook & apache-submarine in jupyter notebook
3. Set two environment variable (SUBMARINE_SERVER_DNS_NAME & SUBMARINE_SERVER_PORT) in notebook container for ease of internal communication between submarine-server and notebook.

### What type of PR is it?
[Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
[SUBMARINE-617](https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-617)

### How should this be tested?
[Travis CI](https://travis-ci.org/github/lowc1012/submarine/builds/729093692)

### Screenshots (if appropriate)
img1

img2

img3

img4

img5

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: dev-support/docker-images/jupyter/Dockerfile [hunk 1]
```diff
@@ -50,22 +44,33 @@ RUN echo "$LOG_TAG Set locale" && \
...
+ENV NB_USER=$NB_USER \
+    NB_UID=$NB_UID \
+    NB_PREFIX=$NB_PREFIX \
+    NB_PORT=$NB_PORT \
+    CONDA_DIR=/opt/conda \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8
+ENV PATH=$CONDA_DIR/bin:$PATH \
+    HOME=/home/$NB_USER
```
Comment (pingsutw) [inline dev-support/docker-images/jupyter/Dockerfile:56]:
```suggestion
    LANGUAGE=en_US.UTF-8 \
    PATH=$CONDA_DIR/bin:$PATH \
    HOME=/home/$NB_USER
```

**Review by jiwq — COMMENTED** [review]
+1 If the comment by @pingsutw had resolved.

**Comment by lowc1012** [comment]
I' m working on that inject submarine server DNS into notebook to easy the  internal communication and SDK initialization. Please do not commit it temporarily. Thanks.

**Comment by lowc1012** [comment]
In the first commit, apache-submarine package will be installed whenever creating notebook server. It spend a lot of time. I suggest that apache-submarine package should be installed before notebook container was running, and spend a lot of time pulling image when creating notebook server first time inevitably.
c.c. @jiwq @pingsutw

**Comment by lowc1012** [comment]
@jiwq @pingsutw 
I set two environment variables (SUBMARINE_SERVER_DNS_NAME and SUBMARINE_SERVER_PORT) in notebook container. Notebook can connect to submarine-server with Submarine SDK without entering the URL.
Please review this PR again. Thanks!

File: dev-support/docker-images/jupyter/start-notebook.sh [hunk 2]
```diff
@@ -42,4 +43,4 @@ if [[ -n "${NB_PREFIX}" ]]; then
   NOTEBOOK_ARGS="--NotebookApp.base_url=${NB_PREFIX} ${NOTEBOOK_ARGS}"
 fi
 
-/bin/bash -c "jupyter notebook ${NOTEBOOK_ARGS}"
+/bin/bash -c "jupyter notebook ${NOTEBOOK_ARGS}"                                                                                                                                                 46,48         Bot
```
Comment (pingsutw) [inline dev-support/docker-images/jupyter/start-notebook.sh:46]: What it is?

**Review by pingsutw — COMMENTED** [review]
other LGTM

File: dev-support/docker-images/jupyter/start-notebook.sh (hunk 2 above)
Comment (pingsutw) [reply dev-support/docker-images/jupyter/start-notebook.sh:46]: 46,48         Bot

**Comment by pingsutw** [comment]
Could you also set the environment used by Jupyter Notebook in https://github.com/apache/submarine/blob/master/docs/database/submarine-data.sql? 
So we don't need to manually create this environment

**Comment by lowc1012** [comment]
@pingsutw Thanks for your review! I pushed a fixed-commit.

**Review by pingsutw — APPROVED** [review]
LGTM

**Review by xunliu — APPROVED** [review]
@lowc1012 Thank you for contribution.
LGTM

**Pull request closed without merging**